infra/src/cli/render-secrets.ts

```diff
@@ -2,79 +2,59 @@ import { generateYamlForEnv } from '../dsl/serialize-to-yaml'
 import { UberChart } from '../dsl/uber-chart'
 import { Envs } from '../environments'
 import { charts } from '../uber-charts/all-charts'
-import { SSM, ResourceGroupsTaggingAPI } from 'aws-sdk'
-import { GetResourcesOutput } from 'aws-sdk/clients/resourcegroupstaggingapi'
+import { SSM } from 'aws-sdk'
+import { GetParametersByPathResult } from 'aws-sdk/clients/ssm'
 
 const API_INITIALIZATION_OPTIONS = {
   region: 'eu-west-1',
   maxRetries: 10,
   retryDelayOptions: { base: 200 },
 }
 
-export const renderSecrets = async (service: string) => {
-  const client = new SSM(API_INITIALIZATION_OPTIONS)
+const client = new SSM(API_INITIALIZATION_OPTIONS)
 
+export const renderSecrets = async (service: string) => {
   const urls: string[] = []
   const uberChart = new UberChart(Envs.dev)
   const services = Object.values(charts).map(
     (chart) => generateYamlForEnv(uberChart, ...chart.dev).services,
   )
-  const secretMap: { [name: string]: string } = {}
+
+  const allDevParamNames = await getAllDevParams()
+
   services.forEach((svc) => {
     Object.entries(svc).forEach(([serviceName, config]) => {
       if (serviceName == service) {
         Object.entries(config.secrets).forEach(([envName, ssmName]) => {
-          secretMap[ssmName] = envName
+          if (allDevParamNames[ssmName]) {
+            console.log(`export ${envName}=${allDevParamNames[ssmName]}`)
+          }
         })
       }
     })
   })
-  // There seem to be only 1 parameter that's labeled as dev?
-  // const allDevParamNames = await getAllDevParamsNames()
-  // const ssmNames = Object.keys(secretMap).filter((ssmName) =>
-  //   allDevParamNames.includes(ssmName),
-  // )
-  const ssmNames = Object.keys(secretMap)
-  while (ssmNames.length) {
-    const params = await client
-      .getParameters({ Names: ssmNames.splice(0, 10), WithDecryption: true })
-      .promise()
-    if (params.Parameters) {
-      params.Parameters.forEach((p) => {
-        if (p.Name) {
-          console.log(`export ${secretMap[p.Name]}=${p.Value}`)
-        }
-      })
-    } else {
-      throw new Error('params.Parameters aws undefined?')
-    }
-    return
-  }
 }
 
-const getAllDevParamsNames = async (): Promise<string[]> => {
-  const client = new ResourceGroupsTaggingAPI(API_INITIALIZATION_OPTIONS)
-  const names = []
+const getAllDevParams = async (): Promise<{ [name: string]: string }> => {
```

Comment:
Aren't there multiple environment variables that have the same name but for different projects (where the value might differ)?

Reply:
This will return a map of `ssm-parameter-name: ssm-parameter-value`. SSM parameter names are unique.

Reply:
Yeah but then get-secrets will strip the environment variable name and load it using dotenv so they'll override each other if the final part of the aws param store variable matches.

Reply:
Final part? What do you mean?

Let's say service A wants to map `/k8s/serviceA/SOME_VAR` to the environment variable `SOME_VAR` and you run `get-secrets serviceA`.

- serviceB wants to map `/k8s/serviceB/SOME_VAR` to the environment variable `NOT_SOME_VAR`. That won't override `SOME_VAR` as `NOT_SOME_VAR != SOME_VAR`
- serviceC wants to map `/k8s/serviceC/NOT_SOME_VAR` to the environment variable `SOME_VAR`. That WILL override `SOME_VAR` in `.env.secrets`, but that's exactly how the script behaved before so I'm not addressing that.

Reply:
Ahh sorry... I somehow assumed that you removed the service argument and you're just fetching every single environment variable, regardless of service.